From 75406e23bf3671e4e0813dd0b698e874554dc306 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:23 -0400
Subject: [usb] net/cdc-acm: fix power management in ioctl

Message-id: <1410887463-20674-192-git-send-email-dzickus@redhat.com>
Patchwork-id: 95414
O-Subject: [RHEL7 PATCH 191/591] cdc-acm: fix power management in ioctl
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 8fdbeb26b5105282e55559b54a660fe7b6eed331
Author: Oliver Neukum <oneukum@suse.de>
Date:   Wed Nov 20 11:35:36 2013 +0100

    cdc-acm: fix power management in ioctl

    An ioctl that does depends on communication with a device should
    prevent suspension of teh device.

    Signed-off-by: Oliver Neukum <oneukum@suse.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/class/cdc-acm.c b/drivers/usb/class/cdc-acm.c
index 3723faf..900f7ff 100644
--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -888,7 +888,13 @@ static int acm_tty_ioctl(struct tty_struct *tty,
   rv = set_serial_info(acm, (struct serial_struct __user *) arg);
   break;
  case TIOCMIWAIT:
+  rv = usb_autopm_get_interface(acm->control);
+  if (rv < 0) {
+   rv = -EIO;
+   break;
+  }
   rv = wait_serial_change(acm, arg);
+  usb_autopm_put_interface(acm->control);
   break;
  case TIOCGICOUNT:
   rv = get_serial_usage(acm, (struct serial_icounter_struct __user *) arg);
-- 
1.7.1